From 0e7bae7d42d3f3874253f05238c38527f0ed48a5 Mon Sep 17 00:00:00 2001
From: Haiyang Zhang <haiyangz@microsoft.com>
Date: Fri, 13 Oct 2017 12:28:05 -0700
Subject: [PATCH 130/156] hv_netvsc: Add initialization of tx_table in
 netvsc_device_add()

tx_table is part of the private data of kernel net_device. It is only
zero-ed out when allocating net_device.

We may recreate netvsc_device w/o recreating net_device, so the private
netdev data, including tx_table, are not zeroed. It may contain channel
numbers for the older netvsc_device.

This patch adds initialization of tx_table each time we recreate
netvsc_device.

Signed-off-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/hyperv/netvsc.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/hyperv/netvsc.c	2017-12-05 16:55:51.890603833 -0800
+++ b/drivers/net/hyperv/netvsc.c	2017-12-05 16:55:51.886603885 -0800
@@ -1252,6 +1252,9 @@ struct netvsc_device *netvsc_device_add(
 	if (!net_device)
 		return ERR_PTR(-ENOMEM);
 
+	for (i = 0; i < VRSS_SEND_TAB_SIZE; i++)
+		net_device_ctx->tx_table[i] = 0;
+
 	net_device->ring_size = ring_size;
 
 	/* Because the device uses NAPI, all the interrupt batching and
